skiplist/src/arena.rs

```diff
@@ -44,13 +45,13 @@ impl Arena {
     pub fn alloc(&self, align: usize, size: usize) -> u32 {
         let align_mask = align - 1;
         // Leave enough padding for align.
-        let size = size + align_mask;
+        let size = (size + align_mask) & !align_mask;
         let offset = self.core.len.fetch_add(size as u32, Ordering::SeqCst);
-        // Calculate the correct align point, it equals to
-        // (offset + align_mask) / align * align.
-        let ptr_offset = (offset as usize + align_mask) & !align_mask;
-        assert!(offset as usize + size <= self.core.cap);
-        ptr_offset as u32
+        if offset as usize + size > self.core.cap {
+            // 0 means there is not enough space to allocate.
+            return 0;
+        }
+        offset as u32
```

Comment:
I thought the returning offset is supposed to be aligned.

Reply:
I'm considering what's the benefits of aligning to the number smaller than `size of pointer`. Should we simply announce that the align must >= sizeof(ptr), to make the code simple.